src/curl.ts

```diff
@@ -0,0 +1,40 @@
+import { BeeRequest, Utils } from '@ethersphere/bee-js'
+import { Readable } from 'stream'
+import { printer } from './printer'
+
+const getDataString = (data?: unknown): string | null => {
+  if (typeof data === 'string') {
+    return data
+  }
+
+  if (data instanceof Readable) {
+    return '<stream>'
+  }
+
+  if (data instanceof Uint8Array) {
+    return '<buffer>'
+  }
+
+  return '<unknown>'
+}
+
+function printCurlCommand(request: BeeRequest): BeeRequest {
+  const params = Object.entries(request.params || {}).filter(([, v]) => v !== undefined)
+  const headers = Object.entries(request.headers || {})
+    .map(([key, value]) => `-H "${key}: ${value}"`)
+    .join(' ')
+  const queryParameters = new URLSearchParams(params as string[][]).toString()
+  const queryString = queryParameters ? '?' + queryParameters : ''
+  const methodString = request.method.toUpperCase() === 'GET' ? '' : ` -X ${request.method?.toUpperCase()}`
+  const dataString = request.data ? ` --data "${getDataString(request.data)}"` : ''
+  const command = `curl${methodString} ${request.url}${queryString} ${headers}${dataString}`
+  printer.print(command)
+
+  return request
+}
+
+export function registerCurlHook(): void {
+  if (typeof process.env.CURL_HOOK_ID === 'undefined') {
```

Comment:
Is this sort of "singleton" design? Only one hook would be registered?

Reply:
Well, kind of.

The tests fail because all tests in a suite register the hook. ^^

I just went with an env variable so the `id` is not completely lost

Reply:
Ahhh, right. I was just wondering that for the normal swarm-cli life-cycle this does not make really sense, but sure for tests do :-) 👍